api/api.go

```diff
@@ -14,38 +14,39 @@ import (
 
 // Strings for VolumeSpec
 const (
-	Name                     = "name"
-	SpecNodes                = "nodes"
-	SpecParent               = "parent"
-	SpecEphemeral            = "ephemeral"
-	SpecShared               = "shared"
-	SpecJournal              = "journal"
-	SpecSharedv4             = "sharedv4"
-	SpecCascaded             = "cascaded"
-	SpecSticky               = "sticky"
-	SpecSecure               = "secure"
-	SpecCompressed           = "compressed"
-	SpecSize                 = "size"
-	SpecScale                = "scale"
-	SpecFilesystem           = "fs"
-	SpecBlockSize            = "block_size"
-	SpecHaLevel              = "repl"
-	SpecPriority             = "io_priority"
-	SpecSnapshotInterval     = "snap_interval"
-	SpecSnapshotSchedule     = "snap_schedule"
-	SpecAggregationLevel     = "aggregation_level"
-	SpecDedupe               = "dedupe"
-	SpecPassphrase           = "secret_key"
-	SpecAutoAggregationValue = "auto"
-	SpecGroup                = "group"
-	SpecGroupEnforce         = "fg"
-	SpecZones                = "zones"
-	SpecRacks                = "racks"
-	SpecRack                 = "rack"
-	SpecRegions              = "regions"
-	SpecLabels               = "labels"
-	SpecPriorityAlias        = "priority_io"
-	SpecIoProfile            = "io_profile"
+	Name                       = "name"
+	SpecNodes                  = "nodes"
+	SpecParent                 = "parent"
+	SpecEphemeral              = "ephemeral"
+	SpecShared                 = "shared"
+	SpecJournal                = "journal"
+	SpecSharedv4               = "sharedv4"
+	SpecCascaded               = "cascaded"
+	SpecSticky                 = "sticky"
+	SpecSecure                 = "secure"
+	SpecCompressed             = "compressed"
+	SpecSize                   = "size"
+	SpecScale                  = "scale"
+	SpecFilesystem             = "fs"
+	SpecBlockSize              = "block_size"
+	SpecHaLevel                = "repl"
+	SpecPriority               = "io_priority"
+	SpecSnapshotInterval       = "snap_interval"
+	SpecSnapshotSchedule       = "snap_schedule"
+	SpecAggregationLevel       = "aggregation_level"
+	SpecDedupe                 = "dedupe"
+	SpecPassphrase             = "secret_key"
+	SpecAutoAggregationValue   = "auto"
+	SpecGroup                  = "group"
+	SpecGroupEnforce           = "fg"
+	SpecZones                  = "zones"
+	SpecRacks                  = "racks"
+	SpecRack                   = "rack"
+	SpecRequestedNodesOptional = "requested_nodes_optional"
```

Comment:
SpecBestEffort="best_effort" // By default this is false. If set provisioning request will succeed even if specified data location parameters could not be satisfied.